server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1471,11 +1471,11 @@ export class AgenticChatController implements ChatHandlers {
         }
 
         // These are errors we want to show custom messages in chat for.
-        if (err.code === 'QModelResponse' || err.code === 'MaxAgentLoopIterations') {
-            this.#features.logging.error(`${err.code}: ${JSON.stringify(err.cause)}`)
+        if (['QModelResponse', 'MaxAgentLoopIterations', 'InputTooLong'].includes(err.code)) {
+            this.#features.logging.error(`${err.code}: ${JSON.stringify(err)} `)
             return new ResponseError<ChatResult>(LSPErrorCodes.RequestFailed, err.message, {
                 type: 'answer',
-                body: err.message,
+                body: err.message + `\n\nRequest ID: ${err.requestId}`,
```

Comment:
I think requestId will only be available for RTS exceptions, please refer to agentic-chat branch implementation: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhispererChat/controllers/chat/controller.ts#L708-L728

Reply:
imo, it's also fine to not show the requestId in the actual chat window, as long as we log it somewhere and it's easy to find

Reply:
Addressed. Only showing if it exists